src/comment.ts

```diff
@@ -129,17 +129,20 @@ const getNoteInfo = (api: Gitlab, mrIid: number | string) =>
     },
   )
 
-const hasChangesetBeenAdded = (
-  changedFilesPromise: ReturnType<MergeRequests['changes']>,
-) =>
-  changedFilesPromise.then(files =>
-    files.changes!.some(
-      file =>
-        file.new_file &&
-        /^\.changeset\/.+\.md$/.test(file.new_path) &&
-        file.new_path !== '.changeset/README.md',
-    ),
-  )
+const hasChangesetBeenAdded = async (
+  changedFilesPromise: ReturnType<MergeRequests['showChanges']>,
+) => {
+  const changedFiles = await changedFilesPromise
+  const changes =
+    'changes' in changedFiles ? changedFiles.changes : changedFiles.data.changes
```

Comment:
Hmm... Why there are two possible paths representing same thing?

Reply:
The `showChanges` function now is a generic function that may return `MergeRequestChangesSchema` or `ExpandedResponse<MergeRequestChangesSchema>` based on the input options. 

Even though we are not providing `{ showExpanded: true }` and the function defaults `E` to be `false`, it cannot be known by the type system if we use `ReturnType` to infer the type of that promise.

Reply:
OK, weird API design though.